stdlib/public/runtime/Casting.cpp

```diff
@@ -2755,6 +2755,50 @@ swift::swift_conformsToProtocol(const Metadata *type,
   goto recur;
 }
 
+/// Return the type metadata for a given type name.
+///
+/// This is a temporary implementation to support some Foundation use cases;
+/// it will only lookup a type that directly implements a protocol as only
+/// those types will have conformance table entries.
+static const Metadata *
+_metadataForName(const std::string &typeName) {
+  auto &C = Conformances.get();
+  const Metadata *foundMetadata = nullptr;
+
+  installCallbacksToInspectDylib();
```

Comment:
Maybe we can factor `installCallbacksToInspectDylib` into the constructor on `Conformances` so that it happens as part of initialization of the conformance cache state. That way it doesn't need to be manually called on both the `conformsToProtocol` and `getTypeByName` paths.